### PR TITLE
8325373: Improve StackCounter error reporting for bad switch cases

### PR DESCRIPTION
**Pull request by asotona**
ClassFile API `StackMapGenerator` attaches print or hex dump of the method to an error message.
However there is no such attachment when the stack maps generation is  turned off.

This patch moves class print/dump to `impl.Util::dumpMethod`, so it is shared by `StackMapGenerator` and `StackCounter` to provide the same level of details in case of an error.

Please review.

Thank you,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325373](https://bugs.openjdk.org/browse/JDK-8325373): Improve StackCounter error reporting for bad switch cases (**Bug** - P4)

### Reviewers
 * [Paul Sandoz](https://openjdk.org/census#psandoz) (@PaulSandoz - **Reviewer**) ⚠️ Review applies to [f59654be](https://git.openjdk.org/jdk/pull/18914/files/f59654be743989b732765950dd34341a18cc6e25)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18914/head:pull/18914` \
`$ git checkout pull/18914`

Update a local copy of the PR: \
`$ git checkout pull/18914` \
`$ git pull https://git.openjdk.org/jdk.git pull/18914/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18914`

View PR using the GUI difftool: \
`$ git pr show -t 18914`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18914.diff">https://git.openjdk.org/jdk/pull/18914.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18914#issuecomment-2072245921)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/impl/Util.java [hunk 1]
```diff
@@ -193,4 +200,41 @@ public static boolean isDoubleSlot(ClassDesc desc) {
...
+                                    b.writeU2(-1);//max stack
+                                    b.writeU2(-1);//max locals
+                                    b.writeInt(bytecode.limit());
+                                    b.writeBytes(bytecode.array(), 0, bytecode.limit());
+                                    b.writeU2(0);//exception handlers
+                                    b.writeU2(0);//attributes
+                                }
+                    }))));
+            ClassPrinter.toYaml(clm.methods().get(0).code().get(), ClassPrinter.Verbosity.TRACE_ALL, dump);
+        } catch (Error | Exception suppresed) {
```
Comment (PaulSandoz) [inline src/java.base/share/classes/jdk/internal/classfile/impl/Util.java:229]: If you like you can replace `suppresed` [sic] with `_`.

**Review by PaulSandoz — APPROVED** [review]
Looks good, just update the copy write date in Util.java.

File: src/java.base/share/classes/jdk/internal/classfile/impl/Util.java (hunk 1 above)
Comment (asotona) [reply src/java.base/share/classes/jdk/internal/classfile/impl/Util.java]: Both fixed, thanks for the review.

**Pull request closed without merging**